examples/15_compression.cpp

```diff
@@ -0,0 +1,328 @@
+/* Copyright 2025 Franz Poeschel
+ *
+ * This file is part of openPMD-api.
+ *
+ * openPMD-api is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * openPMD-api is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with openPMD-api.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <openPMD/openPMD.hpp>
+
+/*
+ * If installed into a folder known to HDF5, then HDF5 will find the filter on
+ * its own. In other contexts, it might become necessary to manually register
+ * the filter into HDF5. For this, link the application against
+ * libblosc2_filter.so and set the below define to true.
+ */
+#define OPENPMD_INIT_BLOSC2_FILTER_MANUALLY false
```

Comment:
Note to self: Pull this logic down in the file, this should not be the first thing users see